pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
but rate will not work. is only for ever increasing counters. 

Grafana counters are like statsd counters (reset each interval). And by using pull Prometheus must pull at the exact same rate as Grafana tries to "push", which might cause issues. 

A better way to support Prometheus might be to expose them via the api, we have a metrics endpoint at /api/metrics  ,  maybe add another one at /api/metrics/prometheus   that supports the prometheus format